### PR TITLE
Add client info

### PR DESCRIPTION
**Pull request by Popcore**
the pull requests 
- adds a `raw_data` property to a 'thing' in order to store the original response generated by the upstream data provider.
- adds a `GetClientInfo` method to the `Client` interface in order to pass to the fetcher any extra information about the client context. This will allow the fetcher to run in behalf of an authenticated user in the case of the Dash API.

### CODE REVIEW

**Comment by smulube** [comment]
I get the `raw_data` thing, but can you bring me up to speed a bit on the usage and purpose of this client info thing a bit please @Popcore and @mdevilliers ?

**Comment by mdevilliers** [comment]
ClientInfo is a mechanism for passing credentials, keys, ect into a fetcher.

**Comment by smulube** [comment]
um, yeah I get that - I was trying to understand how you would actually use it within a fetcher. Should I look at the Dash fetcher, or is there some other snippet you could share showing some example usage?

**Comment by mdevilliers** [comment]
Dash fetcher would be the place to look.

**Comment by smulube** [comment]
@Popcore  - yeah further to our chat yesterday, I don't really like that change to the client, it feels like the wrong place for that data, and the only reason for doing it that way is to avoid having to change the other fetchers (which isn't a terrible goal, but not the best one either).

Wouldn't it be clearer to add a context parameter to `Fetch` although would we also need to pass the same context into `URLS`? Do we need that even? Thinking aloud really

**Comment by Popcore** [comment]
In the specific case if the Dash fetcher there is no need of passing a context to the `URLS` method. It is only needed for authentication when fetching the user's data. But it could become useful for the `URLS` method too in case the fetcher needs to authenticate or it needs any other context specific information in order to do its job.

My feeling would be to add an extra `context` parameter to `Fetch` for now. And add it to the `URLS` method if/when it will be needed.

**Comment by smulube** [comment]
I did read the article the other day that @mdevilliers posted in reading (https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39#.rbn1o0xzs), so I know there is some debate about the proper use of net/context (https://godoc.org/golang.org/x/net/context) but rather than inventing our own context type, this might be a reasonable place to use it no?

In fact there might be an advantage to being able to make Fetch invocations cancelable for the on demand fetching that Mark is working on at the moment, which using the the Context library would give us for free

File: fetcher.go [hunk 1]
```diff
@@ -22,7 +23,7 @@ type Fetcher interface {
 	// also pass in a TimeProvider which is used internally by the fetcher to
 	// record the indexing time of the parser. This is to allow for easier
 	// testing.
-	Fetch(url string, client Client, timeProvider TimeProvider) ([]Thing, error)
+	Fetch(url string, client Client, ctx context.Context, timeProvider TimeProvider) ([]Thing, error)
```
Comment (smulube) [inline fetcher.go]: @Popcore let's be maximum pedantic about this shall we 😉 - those docs say

> Context should be the first parameter, typically named ctx:

**Comment by smulube** [comment]
Also what's happening with the build, something weird going on with glide? How did you add the dep?

**Pull request closed without merging**